eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml

```diff
@@ -3,7 +3,15 @@ steps:
     - pwsh: |
         $filesInCommonDir = git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only -- 'eng/common/*'
 
-        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and ($LASTEXITCODE -eq 0) -and ($filesInCommonDir.Count -gt 0))
+        # Find the default branch of the repo
+        $defaultBranchLine = git remote show origin | Where-Object {$_ -Like "*HEAD branch:*"}
+        $defaultBranchArr =  $defaultBranchLine.Trim().Split()
+        $defaultBranch = $defaultBranchArr[$defaultBranchArr.Count - 1]
+
+        Write-Host "Default Branch: ${defaultBranch}"
+
+        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and
```

Comment:
It might be better to check the source and target branches first and then do the git diff after that condition has been met.